### PR TITLE
SAK-30032 Allow Student to attach files during Peer Review

### PR DESCRIPTION
**Pull request by davidpbauer**
This allows students to add one or more attachments to their reviews when using the Peer Review feature in Assignments.
There is an additional table that is needed for this feature. The conversion scripts for this table can be found in
sakai_11_mysql_conversion.sql or sakai_11_oracle_conversion.sql

### CODE REVIEW

**Comment by buckett** [comment]
Currently has a merge conflict (probably in the conversion scripts).

File: assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/AssignmentPeerAssessmentServiceImpl.java [hunk 1]
```diff
@@ -392,13 +395,53 @@ public Object doInHibernate(Session session) throws HibernateException,
...
+		if(submissionId == null || "".equals(submissionId) || assessorUserId == null || "".equals(assessorUserId)){
+			//return an empty list
+			return new ArrayList<PeerAssessmentAttachment>();
+		}
+		HibernateCallback hcb = session -> {
+            Query q = session.getNamedQuery("findPeerAssessmentAttachmentsByUserAndSubmission");
+            q.setParameter("submissionId", submissionId);
+            q.setParameter("assessorUserId", assessorUserId);
+            return q.list();
+        };
```
Comment (buckett) [inline assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/AssignmentPeerAssessmentServiceImpl.java]: Indentation looks off here.

File: assignment/assignment-api/api/src/java/org/sakaiproject/assignment/api/model/PeerAssessmentItem.java [hunk 2]
```diff
@@ -81,8 +87,24 @@ public boolean isSubmitted() {
...
+
+	public List<PeerAssessmentAttachment> getAttachmentList() {
+		return attachmentList;
+	}
+
+	public void setAttachmentList(List<PeerAssessmentAttachment> attachmentList) {
+		this.attachmentList = attachmentList;
+	}
+
+	public List getAttachmentRefList() {
```
Comment (buckett) [inline assignment/assignment-api/api/src/java/org/sakaiproject/assignment/api/model/PeerAssessmentItem.java]: Can you make the generic? eg `List<Reference>`

File: assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_student_review_edit.vm [hunk 3]
```diff
@@ -385,6 +424,124 @@ function handleEnterKeyPress(ev)
...
+                            <td>
+                                <span>$howManySubAtts</span>
+                            </td>
+                            <td>
+                                <div id="clonableUpload">
+                                    <input type="file" name="upload" class="upload" id="clonableUpload" onchange="javascript:showNotif('submitnotifxxx','post','gradeForm');document.getElementById('gradeForm').action='#toolLinkParam("AssignmentAction" "doAttachUpload" "")';submitform('gradeForm');""/>
+                                    <span id="submitnotifxxx" class="messageProgress" style="visibility:hidden">$tlang.getString("processmessage.file")</span>
+                                </div>
+                            </td>
+                            <!-- <td>
```
Comment (buckett) [inline assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_student_review_edit.vm]: Is this commented out code needed?

**Comment by buckett** [comment]
You're changes to the .vm files look to have a combination of tabs/spaces, this makes them hard to read when the developer doesn't have the same tab size as the author, I know not all of Sakai is very consistent but would you be able to just make sure the new blocks you write are consistent with the majority of the existing file.

Thanks.

**Comment by jonespm** [comment]
If you're going to be able to fix these up these comments and the conflicts by the end of the week this could still be considered for 11.0, otherwise it might need to wait until a future 11 release.

**Comment by davidpbauer** [comment]
Great. Thanks for the feedback. I'll try to find some time tomorrow to revisit this and resolve the conflicts / clean up the code.

**Comment by ottenhoff** [comment]
Hi David, any chance we can get this into master soon to get it into sakai 11?

**Comment by davidpbauer** [comment]
My apologies, I saw the sakai-12 label and assumed there was no rush. I've added this to our next sprint so it will be done sometime in the next two weeks.

**Comment by master-bob** [comment]
@ottenhoff , I've pushed a commit which responds to all the comments and fixes a number of issues.

**Comment by ottenhoff** [comment]
Okay this is looking ok to me

**Pull request merged**